src/flows/pipes/Schedule/ExportTaskButton.tsx

```diff
@@ -15,33 +26,86 @@ import ExportTaskOverlay from 'src/flows/pipes/Schedule/ExportTaskOverlay'
 
 // Utils
 import {event} from 'src/cloud/utils/reporting'
+import {getErrorMessage} from 'src/utils/api'
+import {getOrg} from 'src/organizations/selectors'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
 interface Props {
   text: string
   type: string
   generate?: () => string
+  onCreate?: (task: any) => void
+  disabled?: boolean
 }
 
-const ExportTaskButton: FC<Props> = ({text, type, generate}) => {
+const ExportTaskButton: FC<Props> = ({
+  text,
+  type,
+  generate,
+  onCreate,
+  disabled,
+}) => {
   const {data, range} = useContext(PipeContext)
   const {launch} = useContext(PopupContext)
+  const dispatch = useDispatch()
+  const org = useSelector(getOrg)
 
   const onClick = () => {
+    const query = generate ? generate() : data.query
+
+    if (isFlagEnabled('removeExportModal')) {
+      event('Export Task Modal Skipped', {from: type})
+
+      // we can soft delete the previously connected task by marking the old one inactive
+      if (data?.task?.id) {
+        patchTask({
+          taskID: data.task.id,
+          data: {
+            status: 'inactive',
+          },
+        })
+      }
+
+      postTask({data: {orgID: org.id, flux: query}})
```

Comment:
is there any other meta data we'd want to include onto the data if it's being updated? Like any labels associated with that task?

Reply:
no?